lib/pbench/test/unit/server/test_user_auth.py

```diff
@@ -178,6 +183,49 @@ def test_user_relogin(client, server_config):
                 json={"username": "user", "password": "12345"},
             )
             assert response.status_code == HTTPStatus.OK
+            third_auth_token = response.json["auth_token"]
+            assert (
+                first_auth_token != third_auth_token
+            ), "Third login returned first token"
+            assert (
+                second_auth_token != third_auth_token
+            ), "Third login returned second token"
+
+    @staticmethod
+    def test_login_removes_expired_tokens(client, server_config):
+        """Test to ensure on a new login previously expired tokens are
+        removed."""
+        with client:
+            # user registration
+            resp_register = register_user(
+                client,
+                server_config,
+                username="me",
+                firstname="first name",
+                lastname="last name",
+                email="me@domain.com",
+                password="12345",
+            )
+            assert resp_register.status_code == HTTPStatus.CREATED
+
+            user = User.query(username="me")
+            # Add 3 expired tokens, values are in "days" old
+            for age in (1, 2, 4):
+                token = ActiveToken(
+                    f"token-that-is-{age:d}-days-old",
+                    datetime.now(timezone.utc) - timedelta(days=age),
+                )
+                user.add_token(token)
+
+            response = client.post(
+                f"{server_config.rest_uri}/login",
+                json={"username": "me", "password": "12345"},
+            )
```

Comment:
Does this strike anyone else as weird?  We're making "local" changes to the database, then posting a RESTful request to a "remote" server, and then looking for changes in the "local" database.  🥴  (I get that the "remote" service is also accessing the "local" database, but...still....)

Reply:
"Weird" in the sense that our API "unit tests" really *aren't* "unit" in any meaningful sense as they rely on both `Flask` and `sqlite3`, sure. However, within this quirk-filled context, it's perfectly fine.